For_Andrey/task_3.py

```diff
@@ -0,0 +1,244 @@
+    DECREASE_VALUE_PETROL_PER_THOUSAND_KM = 95
+    FIRST_ENGINE_TYPE = 'diesel'
+    SECOND_ENGINE_TYPE = 'petrol'
+
+    def __init__(self, engine: str, gas_tank: int):
+        assert engine in ('diesel', 'petrol'), f'Incorrect engine type: {engine}'
+        assert gas_tank in (60, 75), f'Incorrect gas_tank type: {gas_tank}'
+        self.engine = engine
+        self.gas_tank = gas_tank
+        self.mileage = randint(55_000, 286_000)
+        """
+        :param engine: type engine in the car
+        :param gas_tank: volume gas_tank in the car
+        """
+
+    def number_of_repairs(self) -> int:
+        """
+        Gets engine type and calculate repairs
+
+        :return:number of repairs for car
+        """
+        if self.engine == Car.FIRST_ENGINE_TYPE:
+            number_of_repairs = self.mileage // Car.MILEAGE_BEFORE_REPAIR_DIESEL
+        else:
+            # For petrol car
+            number_of_repairs = self.mileage // Car.MILEAGE_BEFORE_REPAIR_PETROL
+        return number_of_repairs
+
+    def repair_cost(self) -> int:
+        """
+        Calculate repair cost depending on engine type
+
+        :return:repair cost for car
+        """
+        number_of_repairs = Car.number_of_repairs(self)
+        if self.engine == Car.FIRST_ENGINE_TYPE:
+            repair_cost = Car.REPAIR_FOR_DIESEL_CAR * number_of_repairs
+        else:
+            # For petrol car
+            repair_cost = Car.REPAIR_FOR_PETROL_CAR * number_of_repairs
+        return repair_cost
+
+    def fuel_cost(self) -> Union[int, float]:
+        """
+        Calculate fuel cost for cars after mileage
+
+        :return:fuel cost with module round
+        """
+        mileage = self.mileage
+        total_fuel_cost = 0
+        kef = 0
+        if self.engine == Car.FIRST_ENGINE_TYPE:
+            while True:
+                if mileage > 1000:
+                    fuel_cost = Car.FUEL_CONSUMPTION_DIESEL_ON_HUNDRED_KM * 1000 * Car.DIESEL_FUEL_COST
+                    total_fuel_cost += fuel_cost + (fuel_cost * kef)
+                    kef += 0.01
+                    mileage -= 1000
+                else:
+                    fuel_cost = Car.FUEL_CONSUMPTION_DIESEL_ON_HUNDRED_KM * mileage * Car.DIESEL_FUEL_COST
+                    total_fuel_cost += fuel_cost + (fuel_cost * kef)
+                    break
+        else:
+            # For petrol car
+            while True:
+                if mileage > 1000:
+                    fuel_cost = Car.FUEL_CONSUMPTION_PETROL_ON_HUNDRED_KM * 1000 * Car.PETROL_FUEL_COST
+                    total_fuel_cost += fuel_cost + (fuel_cost * kef)
+                    kef += 0.01
+                    mileage -= 1000
+                else:
+                    fuel_cost = Car.FUEL_CONSUMPTION_PETROL_ON_HUNDRED_KM * mileage * Car.PETROL_FUEL_COST
+                    total_fuel_cost += fuel_cost + (fuel_cost * kef)
+                    break
+        return round(total_fuel_cost)
```

Comment:
too huge and unreadable method. Please create a separate method to calculate the cost of fuel, which would receive consumption and fuel cost as params.
for petrol and desel you have duplicated code

Reply:
fixed